akka-http-core/src/main/scala/akka/http/impl/settings/ClientConnectionSettingsImpl.scala

```diff
@@ -24,7 +24,8 @@ private[akka] final case class ClientConnectionSettingsImpl(
   requestHeaderSizeHint: Int,
   websocketRandomFactory: () ⇒ Random,
   socketOptions: immutable.Seq[SocketOption],
-  parserSettings: ParserSettings)
+  parserSettings: ParserSettings,
+  redirectSettings: akka.http.scaladsl.settings.ClientAutoRedirectSettings)
```

Comment:
The settings are very smartly structured to allow us to use case classes while retaining public API binary compatibility, you'll notice that:

```scala
/** INTERNAL API */
private[akka] final case class ClientConnectionSettingsImpl(
```

So noone is supposed to touch this class => we can change it and filter out MiMa warnings that it has about it.